components/org.wso2.carbon.identity.conditional.auth.functions.http/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/http/AbstractHTTPFunction.java

```diff
@@ -0,0 +1,174 @@
+                if (responseCode >= 200 && responseCode < 300) {
+                    outcome = Constants.OUTCOME_SUCCESS;
+                    String jsonString = EntityUtils.toString(response.getEntity());
+                    JSONParser parser = new JSONParser();
+                    json = (JSONObject) parser.parse(jsonString);
+                } else {
+                    outcome = Constants.OUTCOME_FAIL;
+                }
+
+            } catch (IllegalArgumentException e) {
+                LOG.error("Invalid Url: " + epUrl, e);
+                outcome = Constants.OUTCOME_FAIL;
+            } catch (ConnectTimeoutException e) {
+                LOG.error("Error while waiting to connect to " + epUrl, e);
+                outcome = Constants.OUTCOME_TIMEOUT;
+            } catch (SocketTimeoutException e) {
+                LOG.error("Error while waiting for data from " + epUrl, e);
+                outcome = Constants.OUTCOME_TIMEOUT;
+            } catch (IOException e) {
+                LOG.error("Error while calling endpoint. ", e);
+                outcome = Constants.OUTCOME_FAIL;
+            } catch (ParseException e) {
+                LOG.error("Error while parsing response. ", e);
+                outcome = Constants.OUTCOME_FAIL;
+            }
+
+            asyncReturn.accept(context, json != null ? json : Collections.emptyMap(), outcome);
+        });
+        JsGraphBuilder.addLongWaitProcess(asyncProcess, eventHandlers);
+    }
+
+    private boolean isValidRequestDomain(URI url) {
+
+        if (url == null) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Provided url for domain restriction checking is null");
+            }
+            return false;
+        }
+
+        if (allowedDomains.isEmpty()) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("No domains configured for domain restriction. Allowing url by default. Url: "
+                        + url.toString());
+            }
+            return true;
+        }
+
+        String domain = getParentDomainFromUrl(url);
+        if (StringUtils.isEmpty(domain)) {
+            LOG.error("Unable to determine the domain of the url: " + url.toString());
+            return false;
+        }
+
+        if (allowedDomains.contains(domain)) {
+            return true;
+        }
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Domain: " + domain + " extracted from url: " + url.toString() + " is not available in the " +
+                    "allowed domain list: " + StringUtils.join(allowedDomains, ','));
+        }
+
+        return false;
+    }
+
+    private String getParentDomainFromUrl(URI url) {
+
+        String parentDomain = null;
+        String domain = url.getHost();
+        String[] domainArr = null;
+        if (domain != null) {
+            domainArr = StringUtils.split(domain, DOMAIN_SEPARATOR);
+        }
+
+        if (domainArr != null && domainArr.length != 0) {
```

Comment:
since `domainArr` is populated only if the `domain != null` condition is true, shall we move this if block also inside the previous if block. and we can get rid of `domainArr != null` condition since `StringUtils.split` method will return only if the provided string is null.

Reply:
fixed